benchexec/tools/depthk.py

```diff
@@ -0,0 +1,159 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import subprocess
+import os
+import benchexec.util as Util
+import benchexec.tools.template
+import benchexec.result as result
+import re
+
+
+class Tool(benchexec.tools.template.BaseTool):
+
+    """
+    This class serves as tool adaptor for DepthK (www.esbmc.org)
+    Autor: Williame Rocha - williame.rocha10@gmail.com - Federal University of Amazonas, Brazil.
+    """
+
+    REQUIRED_PATHS = [
+        'depthk.py',
+        'depthk-wrapper.sh',
+        'esbmc',
+        '__init__.py',
+        'modules',
+        'tokenizer',
+        ]
+
+    def executable(self):
+
+        # Relative path to depthk wrapper
+
+#        return Util.find_executable('../../esbmc+depthk/depthk-wrapper.sh')
+
+        return Util.find_executable('depthk-wrapper.sh'
+                                    )
+
+    def working_directory(self, executable):
+        executableDir = os.path.dirname(executable)
+        return executableDir
+
+    def environment(self, executable):
+        return {'additionalEnv': {'PATH': ':.'}}
+
+    def version(self, executable):
+        workingDir = self.working_directory(executable)
+
+        version = subprocess.Popen([workingDir + '/depthk.py',
+                                   '--version'],
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.STDOUT).stdout.readline().decode()
+        return version
+
+    def name(self):
+        return 'DepthK'
+
+    def cmdline(
+        self,
+        executable,
+        options,
+        tasks,
+        propertyfile,
+        rlimits,
+        ):
+
+        assert len(tasks) == 1, 'only one sourcefile supported'
+        assert propertyfile, 'property file required'
+        sourcefile = tasks[0]
+        return [executable] + options + ['-c', propertyfile, sourcefile]
+
+    def determine_result(
+        self,
+        returncode,
+        returnsignal,
+        output,
+        isTimeout,
+        ):
+
+        if len(output) <= 0:
+            return
+
+        output = output[-1].strip()
+        status = ''
+
+        if 'TRUE' in output:
+            status = result.RESULT_TRUE_PROP
+        elif 'no-overflow' in output:
+            status = result.RESULT_FALSE_OVERFLOW
+        elif 'valid-deref' in output:
+            status = result.RESULT_FALSE_DEREF
+        elif 'valid-memtrack' in output:
+            status = result.RESULT_FALSE_MEMTRACK
+        elif 'FALSE' in output:
+            status = result.RESULT_FALSE_REACH
+        else:
+            status = result.RESULT_UNKNOWN
+
+        # else:
+        #    status = result.RESULT_ERROR
+
+        return status
```

Comment:
This method should not return an empty string or `None`. Please change that `result.RESULT_ERROR` is returned if no result can be determined (as it was before).